### DISCUSSION TITLE
How to extend the styles applied by the toolkit with UseVsTheme

### DISCUSSION THREAD

**Discussion by pascalgyger** [VsixCommunity/Community.VisualStudio.Toolkit#135] [Q&A]
For our extension we would like to use the VS themes provided by the toolkit, but would like to extend the styles (e.g. adding tooltip on validation error to a textbox with a style.trigger).
Is there a way we can extend those styles for example with BasedOn? I noticed that the themes from the toolkit don't define their one keys so that we could refer to those styles. Or should we just refer to the same keys that are used in the ThemeResources.xaml in the toolkit (shell:VsResourceKeys.TextBoxStyleKey)?

**Comment by awschristou** [comment]
@reduckted have you got any ideas on this one? I am in a similar situation where I would like to use some data triggers, but this reverts the control's style to the non-themed base style.

**Reply by reduckted** [reply]
It's currently not possible because of how the `ResourceDictionary` is being loaded. We have to replace the assembly names that are specified in the `xmlns` attributes before the XAML can be loaded, because different versions of the toolkit reference different assemblies (specifically the `Microsoft.VisualStudio.Shell.14.0` or `Microsoft.VisualStudio.Shell.15.0` assembly, I think).

I do have a solution to avoid the special loading of the `ResourceDictionary`, and I've just about got it working. I'm just ironing out some bugs with it, and hope to submit a pull request in the next few days.

**Reply by reduckted** [reply]
I've just created #316, which should now allow you to customize the styles that are defined in the toolkit.
